runtime/include/vec_u8.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <stdint.h>
+#include <stdlib.h>
+
+struct vec_u8 {
```

Comment:
The intention here is that this header will be gradually enhanced to add typical push, pop APIs that realloc the buffer as required. I imagine this will probably end up wrapped as a macro to enable generics, but I've also considered keeping the buffer uint8_t[] and instead add a type tag that gets populated from a C11 _Generic macro.